cmake/SpectreCheckTargetDependencies.cmake

```diff
@@ -0,0 +1,652 @@
+  unset(SPECTRE_INCLUDES)
+  foreach(INCLUDE ${_RAW_SPECTRE_INCLUDES})
+    string(REPLACE "\n#include \"" "" INCLUDE "${INCLUDE}")
+    string(REPLACE "\"" "" INCLUDE "${INCLUDE}")
+    list(APPEND SPECTRE_INCLUDES ${INCLUDE})
+  endforeach(INCLUDE ${_RAW_SPECTRE_INCLUDES})
+  # "Return" SPECTRE_INCLUDE by setting it in PARENT_SCOPE
+  set(SPECTRE_INCLUDES ${SPECTRE_INCLUDES} PARENT_SCOPE)
+endfunction(_extract_spectre_includes FILE_NAME)
+
+# Get the list of 3rd party library includes for the target `TARGET_NAME`.
+#
+# Sets the variable `SPECTRE_INCLUDES` in parent scope to return the result
+function(_extract_tpl_includes FILE_NAME)
+  file(READ ${FILE_NAME} FILE_CONTENTS)
+  string(REGEX MATCHALL "\n#include <[^\n]+>"
+    _RAW_SPECTRE_INCLUDES ${FILE_CONTENTS})
+  unset(SPECTRE_INCLUDES)
+  foreach(INCLUDE ${_RAW_SPECTRE_INCLUDES})
+    string(REPLACE "\n#include <" "" INCLUDE "${INCLUDE}")
+    string(REPLACE ">" "" INCLUDE "${INCLUDE}")
+    list(APPEND SPECTRE_INCLUDES ${INCLUDE})
+  endforeach(INCLUDE ${_RAW_SPECTRE_INCLUDES})
+  # "Return" SPECTRE_INCLUDE by setting it in PARENT_SCOPE
+  set(SPECTRE_INCLUDES ${SPECTRE_INCLUDES} PARENT_SCOPE)
+endfunction(_extract_tpl_includes FILE_NAME)
+
+# Make the path of `HEADER_FILE` which belongs to the target `TARGET_NAME`
+# an absolute path.
+#
+# Sets the variable `ABS_HEADER_FILE` in parent scope to return the result
+function(_make_header_path_absolute TARGET_NAME HEADER_FILE)
+  get_property(
+    _INCLUDE_DIR
+    TARGET ${TARGET_NAME}
+    PROPERTY INTERFACE_INCLUDE_DIRECTORIES
+    )
+  if(IS_ABSOLUTE ${HEADER_FILE})
+    set(_ABS_HEADER_FILE ${HEADER_FILE})
+  else()
+    set(_ABS_HEADER_FILE "${_INCLUDE_DIR}/${HEADER_FILE}")
+  endif()
+  # "Return" ABS_HEADER_FILE by setting it in PARENT_SCOPE
+  set(ABS_HEADER_FILE ${_ABS_HEADER_FILE} PARENT_SCOPE)
+endfunction(_make_header_path_absolute TARGET_NAME HEADER_FILE)
+
+# Gets the dependencies for a list of includes.
+#
+# Example usage:
+#
+#   _get_deps_for_spectre_includes(
+#     ${TARGET_NAME}
+#     "${LIST_OF_ALL_TARGETS}"
+#     "${SPECTRE_INCLUDES}"
+#     "${_HEADER_FILES}"
+#     ${RECURSE_INCLUDES}
+#     )
+#
+# Arguments:
+#
+# TARGET_NAME: the target for which to check the link libraries/dependencies
+# LIST_OF_ALL_TARGETS: a list of all the targets that might be dependencies
+#                      of the target ${TARGET_NAME}
+# SPECTRE_INCLUDES_OF_FILE:
+#     a list of the includes of a file for which to get the dependencies
+# TARGET_HEADER_FILES:
+#     a list of header files for the target ${TARGET_NAME}
+# RECURSE_THROUGH_INCLUDES:
+#     if TRUE then any include in any of ${SPECTRE_INCLUDES_OF_FILE}
+#     that is also in the list ${TARGET_HEADER_FILES} has its dependencies
+#     also added.
+#
+# Since the recursion will cause an infinite loop if there are cyclic includes
+# we limit the recursion depth to 20, at which point an error message is
+# displayed that there may be cyclic dependencies
```

Comment:
It would be helpful, as you've done in many other cases, to note how this returns by setting something at parent scope.